### PR TITLE
tasks for loops, for each

### PR DESCRIPTION
**Pull request by tetiana-qa7**

### CODE REVIEW

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StrongestRobot.java
Comment (rodikno) [inline Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StrongestRobot.java:1]: I like the solution and the progress indication, well done :)

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     */
+
+    //for each
+
+    public static void main(String[] args) {
+        int[] studentIDs = {12345, 67890, 54321, 98765, 24680};
+        int searchID = 54321;
+        boolean isFound = false;
+
+        for (int i : studentIDs) {
```
Comment (rodikno) [inline Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java]: valid, but better practice would be to call it smth like `int studentId : studentIDs` which would better represent what it holds inside it during the iteration

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        if (!isFound) {
+            System.out.println("This student is not found");
+        }
+
+
+        //for
+
+
+        for (int j = 0; j < studentIDs.length; j++) {
```
Comment (rodikno) [inline Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java]: There's a small bug in the body of this loop

Hint: if the desired Id is having index 1, 3, 5, 7, ... -> the script will print "Not found" even if the ID is there inside the array

Comment (rodikno) [reply Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java]: image

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/WaterIntakeTracker.java
Comment (rodikno) [inline Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/WaterIntakeTracker.java:1]: Good job!

**Review by rodikno — CHANGES_REQUESTED** [review]
Please fix the issues according to the comments and mark conversations as resolved - we'll go for the 2nd review round, good luck!

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java (hunk 1 above)
Comment (tetiana-qa7) [reply Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java]: thanks

File: Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+        //for
+
+
+        for (int i = 0; i < studentIDs.length; i++) {
+            if (studentIDs[i] == searchID) {
+                System.out.println("----This student is found");
+                isFound = true;
+                break;
```
Comment (rodikno) [inline Homework_7_Java_Arrays_Loops/JavaArraysAndLoops/src/StudentSearch.java:41]: Looks good now, approved :)

**Review by rodikno — APPROVED** [review]

**Pull request merged**